src/Disks/IO/ReadBufferFromRemoteFSGather.cpp

```diff
@@ -95,7 +95,30 @@ ReadBufferFromRemoteFSGather::ReadBufferFromRemoteFSGather(
     , blobs_to_read(blobs_to_read_)
     , settings(settings_)
     , log(&Poco::Logger::get("ReadBufferFromRemoteFSGather"))
+    , query_id(getQueryId())
+    , enable_cache_log(!query_id.empty() && settings.enable_filesystem_cache_log)
+{
+    with_cache = settings.remote_fs_cache
+        && settings.enable_filesystem_cache
+        && (!IFileCache::isReadOnly() || settings.read_from_filesystem_cache_if_exists_otherwise_bypass_cache);
+}
+
+
+void ReadBufferFromRemoteFSGather::appendFilesystemCacheLog()
 {
+    FilesystemCacheLogElement elem
+    {
+        .event_time = std::chrono::system_clock::to_time_t(std::chrono::system_clock::now()),
+        .query_id = query_id,
+        .source_file_path = getFileName(),
+        .file_segment_range = { 0, total_bytes_read },
+        .read_type = FilesystemCacheLogElement::ReadType::READ_FROM_FS_BYPASSING_CACHE,
+        .file_segment_size = total_bytes_read,
+        .cache_attempted = false,
```

Comment:
Isn't it a bit confusing that we have a column which is always the same value?

Why not add this log for all the read files?

Reply:
> hm, but it is true in case of CachedReadBufferFromRemoteFS and false otherwise, not sure what is a better way to tell whether cache was attempted/checked at all.

Oh I didn't see the diff for the `CachedReadBufferFromRemoteFS`, looks good